src/server/getters/get-trading-history.ts

```diff
@@ -38,10 +39,10 @@ export async function getTradingHistory(db: Knex, augur: Augur, params: t.TypeOf
       "timestamp",
       "tradeGroupId",
     ]), {
-      type: trade.orderType === "buy" ? "sell" : "buy",
+      type: isMaker ? trade.orderType : (trade.orderType === "buy" ? "sell" : "buy"), // in the db, trades.orderType is from perspective of order creator/maker, so if we just use value of trades.orderType then the UITrade.type will always be from perspective of order creator/maker. As a UX concern, we want to default to the perspective of filler/taker, so we flip buy/sell unless isMaker.
       price: trade.price.toString(),
       amount: trade.amount.toString(),
-      maker: params.account == null ? null : params.account === trade.creator,
+      maker: isMaker,
```

Comment:
so for generic market trade history request (no login account passed) the perspective will be from maker? Just want to make sure.

Reply:
IsMaker is set above to be the same expression, on Line 29 -- so if not account is passed in IsMaker will still return null.